torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -79,16 +101,43 @@ def _create_label(self, edge_label_index, edge_label):
 
         edge_label = torch.cat([pos_edge_label, neg_edge_label], dim=0)
 
-        return edge_label_index, edge_label
+        return edge_label_index, edge_label, edge_time
+
+    def _modify_node_time(self, edge_label_index, edge_time):
+        """For edges in a batch replace `src` and `dst`
+        node times by the max across all edge times."""
+        def update_time(index, input_type):
+            new_node_time, _ = scatter_min(edge_time, index,
+                                           dim_size=self.num_src_nodes)
+            index_unique = index.unique()
+            self.node_time_dict[input_type][index_unique] = torch.min(
+                new_node_time[index_unique],
+                self.node_time_dict[input_type][index_unique])
+
+        update_time(edge_label_index[0], self.input_type[0])
+        update_time(edge_label_index[1], self.input_type[-1])
+
+    def _reset_node_time(self, edge_label_index):
```

Comment:
No need to reset when we simply pass in a temporary dictionary to `hetero_sparse_neighbor_sample` without modifying the underlying attribute. Sounds cleaner to me.

Reply:
I didn't fully get you here. 
Do you mean we  create a copy of `self.node_time_dict[input_type]` and update that tensor in `update_time`.

Reply:
I don't think we should modify the `node_time_dict` property here. Then there is also no need to modify reset the property at all. Instead, let us shallow-copy `node_time_dict`, modify it, and input the modified copy to the sampler:
```python
def __call__(...):
    # Modify node_time_dict
    node_time_dict = copy.copy(node_time_dict)
    node_time_dict[...] = ... # make changes
    self._hetero_sample(..., node_time_dict=node_time_dict)
```

Reply:
Thanks for this suggestions. As you mentioned this is needed for correct behaviour when `num_workers>0`.